Scripts/DCS-BIOS/lib/Protocol.lua

```diff
@@ -75,75 +75,72 @@ function BIOS.protocol.endModule()
-					if maskStr ~= "" then line = line .. ", " .. maskStr end
-					if shiftByStr ~= "" then line = line .. ", " .. shiftByStr end
-
-					if not existingDefines[full_identifier] then
-						table.insert(lineOrder, full_identifier)
-					end
-
-					existingDefines[full_identifier] = line
-
-					-- Additional line with only the address and _ADDR suffix
-					if addressStr ~= "" then
-						local addressOnlyIdentifier = full_identifier .. "_ADDR"
-						local addressOnlyLine = "#define " .. addressOnlyIdentifier .. " " .. addressStr
-
-						if not existingDefines[addressOnlyIdentifier] then
-							table.insert(lineOrder, addressOnlyIdentifier)
+		for moduleName, moduleBeingDefined in pairs(exportModules) do
+			for _, category in pairs(moduleBeingDefined.documentation) do
+				for identifier, args in pairs(category) do
+					local outputs = args.outputs or {}
+					for _, output in pairs(outputs) do
+						local address_identifier = (output.address_mask_shift_identifier or BIOS.util.addressDefineIdentifier(moduleName, identifier)) .. "_A"
+						local address_mask_identifier = (output.address_mask_shift_identifier or BIOS.util.addressDefineIdentifier(moduleName, identifier)) .. "_AM"
+						local address_mask_shiftby_identifier = output.address_mask_shift_identifier or BIOS.util.addressDefineIdentifier(moduleName, identifier)
+						local address = output.address and string.format("0x%04X", output.address) or ""
+						local mask = output.mask and string.format("0x%04X", output.mask) or ""
+						local shift_by = output.shift_by and tostring(output.shift_by) or ""
+
+						local address_line = "#define " .. address_identifier .. " " .. address
+						local address_mask_line = "#define " .. address_mask_identifier .. " " .. address .. ", " .. mask
+						local address_mask_shiftby_line = "#define " .. address_mask_shiftby_identifier .. " " .. address .. ", " .. mask .. ", " .. shift_by
+
+						-- #define lines based on type
+						if output.type == "integer" then
+							addLine(address_mask_shiftby_identifier, address_mask_shiftby_line)
+							if output.max_value == 1 then
+								addLine(address_mask_identifier, address_mask_line)
+							elseif output.max_value == 65535 then
+								addLine(address_identifier, address_line)
+							end
+						end
+						if output.type == "string" then
+							addLine(address_identifier, address_line)
 						end
-
-						existingDefines[addressOnlyIdentifier] = addressOnlyLine
 					end
 				end
 			end
 		end
 
-		-- Write the updated content to the file
-		local address_header_file, err = io.open(path, "w")
+		-- Sort the identifiers before writing
+		local sortedIdentifiers = {}
+		for identifier in pairs(addresses) do
+			table.insert(sortedIdentifiers, identifier)
+		end
+		table.sort(sortedIdentifiers)
+
+		-- Write the header file
+		local address_header_file, err = io.open(lfs.writedir()..[[Scripts\DCS-BIOS\doc\Addresses.h]], "w")

```

Comment:
With the recent changes to allow local compiles and for this to run a local compile in github actions, we should be using the forward slash as our path separator as it works on all operating systems. I have no idea where this file would be written to on a unix system, but probably not where we want lol
```suggestion
		local address_header_file, err = io.open(lfs.writedir()..[[Scripts/DCS-BIOS/doc/Addresses.h]], "w")

```